atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/SnapshotStore.java

```diff
@@ -27,15 +27,21 @@
  * and and removes snapshots when they are no longer referenced by any live start timestamps.
  */
 public interface SnapshotStore {
+    /**
+     * Stores a snapshot for a given sequence and set of timestamps. Calls with the same sequence but a different
+     * snapshot will overwrite the snapshot - this means that multiple calls to {@link #getSnapshot(StartTimestamp)}
+     * may return different snapshots over time for the same start timestamp. However, these newer snapshots should
+     * just be more complete versions of past snapshots, and thus maintains correctness (the timestamp could
+     * correctly receive any one of the snapshots, and the only factor that changes is how much is cached, not the
+     * values themselves).
```

Comment:
It _is_ possible for me to write a check that every overwriting snapshot contains the same locked cells as the old one, but I think that this is probably more expensive than it is worth (O(N) in number of values cached, which is probably more trouble than it is worth).

Reply:
This is fine, maybe wording should reflect that the implementation should satisfy this rather than claiming that the interface magically does it :P

Reply:
Or rather that the usage must be such to preserve correctness